apps/docs/src/app/platform/component-docs/platform-forms/platform-datetime-picker/platform-datetime-picker-examples/platform-datetime-picker-reactive-example.component.ts

```diff
@@ -15,6 +15,13 @@ export class PlatformDatetimePickerReactiveExampleComponent {
     requiredDateValidator: ValidatorFn[] = [Validators.required];
 
     onSubmit(): void {
+        Object.keys(this.datetimePickerForm.controls).forEach((field) => {
+            const control = this.datetimePickerForm.get(field);
+            control.markAsTouched({ onlySelf: true });
+        });
+        if (!this.datetimePickerForm.get('withAllowNull').value) {
+            this.datetimePickerForm.get('withAllowNull').setErrors({ required: true });
+        }
```

Comment:
Not sure about this fix...
The question is what a general strategy we should follow per design.
If the design says that once a form is submitted all invalid controls should become in the invalid state then this logic should be incapsulated withing the platform form implementation.
@fkolar What do you think?

Reply:
I agree. Currently, on form submit I see different things happening on different components. I am not sure what is the 'correct' behaviour that we can agree on. This fix was done only to address the raised issue 😓

Reply:
I see, based on [this guide](https://experience.sap.com/fiori-design-web/form-field-validation/#on-create-or-save) we do need to validate control on submit event.
I would suggest to skip this fix and create more precise issue describing a root of the problem and fix it in scope of `platform form` not in a scope of this particular example.
@fkolar could you judge us please

Reply:
Also to note that from the guide and UI5's [implementation](https://sapui5.hana.ondemand.com/#/entity/sap.ui.core.message.MessageManager/sample/sap.ui.core.sample.MessageManager.BasicMessages), it seems a message popover would also be needed that shows up the validation messages separately as well.
![image](https://user-images.githubusercontent.com/53509521/100968122-c5632200-3556-11eb-88ef-059ac8f825b1.png)

Reply:
> Also to note that from the guide and UI5's [implementation](https://sapui5.hana.ondemand.com/#/entity/sap.ui.core.message.MessageManager/sample/sap.ui.core.sample.MessageManager.BasicMessages), it seems a message popover would also be needed that shows up the validation messages separately as well.
> ![image](https://user-images.githubusercontent.com/53509521/100968122-c5632200-3556-11eb-88ef-059ac8f825b1.png)

I think it's the feature for future releases and not important now.